cloudify_starlingx_sdk/common.py

```diff
@@ -67,8 +95,16 @@ def merge_configs(self, config):
         os_kwargs = config.pop('os_kwargs', {})
         config.update(os_kwargs)
         for key in ['os_auth_url', 'auth_url']:
-            if key in config:
-                config[key] = self.cleanup_auth_url(config[key])
+            if key not in config:
+                continue
+            # Make sure that we are sending a useful URL.
+            config[key] = self.cleanup_auth_url(config[key])
+            # Check that https is used appropriately.
+            if not (config.get('insecure', False) or
+                    'cacert' in config or
+                    'os_cacert' in config) and \
```

Comment:
if we have cacert we do  want to use https no?

Reply:
It's if we don't have `if not`

Reply:
oh my mistake